src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -1,124 +1,241 @@
-﻿using System.Collections.Generic;
-using System.Linq;
+﻿using System;
+using System.Collections.Generic;
 using Godot;
 
 public class ChemoreceptorUpgradeGUI : VBoxContainer, IOrganelleUpgrader
 {
     [Export]
-    public NodePath? CompoundsPath;
+    public NodePath? TargetTypesPath;
+
+    [Export]
+    public NodePath CompoundsPath = null!;
+
+    [Export]
+    public NodePath CompoundsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPath = null!;
+
+    [Export]
+    public NodePath SpeciesLabelPath = null!;
 
     [Export]
     public NodePath MaximumDistancePath = null!;
 
     [Export]
     public NodePath MinimumAmountPath = null!;
 
+    [Export]
+    public NodePath MinimumAmountLabelPath = null!;
+
     [Export]
     public NodePath ColourPath = null!;
 
 #pragma warning disable CA2213
+    private OptionButton targetTypes = null!;
     private OptionButton compounds = null!;
+    private Label compoundLabel = null!;
+    private OptionButton species = null!;
+    private Label speciesLabel = null!;
```

Comment:
And these are the related variables that should also be renamed if my previous rename comments are accepted.

Reply:
The above variables were renamed, but these weren't yet to match the new names of the related variables.